test/fail_compilation/fail22133.d

```diff
@@ -0,0 +1,24 @@
+// https://issues.dlang.org/show_bug.cgi?id=22133
+/*
+TEST_OUTPUT
+---
+fail_compilation/fail22133.d(16): Error: `s.popFront()()` has no effect
```

Comment:
As a follow-up, we could add a `TOK.dotTemplateDeclaration` case in `discardValue()` and offer a better error message.